src/topics/unread.js

```diff
@@ -273,12 +293,29 @@ module.exports = function (Topics) {
 			return false;
 		}
 		const [topicScores, userScores] = await Promise.all([
-			Topics.getTopicsFields(tids, ['tid', 'lastposttime']),
+			Topics.getTopicsFields(tids, ['tid', 'lastposttime', 'timestamp']),
 			db.sortedSetScores(`uid:${uid}:tids_read`, tids),
 		]);
 
-		tids = topicScores.filter((t, i) => t.lastposttime && (!userScores[i] || userScores[i] < t.lastposttime))
-			.map(t => t.tid);
+		tids = topicScores.filter(filterAlreadyRead).map(t => t.tid);
+
+		function filterAlreadyRead(topic, index) {
+			if (!topic.lastposttime) {
+				return false;
+			}
+
+			const userReadTime = userScores[index];
+			if (!userReadTime) {
+				return true;
+			}
+
+			const isMainPost = topic.timestamp === topic.lastposttime;
```

Comment:
If you are trying to filter out scheduled topics maybe you can use the 'scheduled` field?

Reply:
This part is completely changed too.